### PR TITLE
CLDR-17137 Add --license-file to Ldml2json for overriding the license included in the binary

### PR DESCRIPTION
**Pull request by Manishearth**
CLDR-17137

- [x] This PR completes the ticket.

ALLOW_MANY_COMMITS=true

### CODE REVIEW

**Review by srl295 — APPROVED** [review]
LGTM

Need to run `mvn spotless:apply`

**Review by srl295 — APPROVED** [review]

**Comment by srl295** [comment]
FYI @btangmu 

@Manishearth are you able to merge it yourself?

**Comment by Manishearth** [comment]
Nope!

**Comment by Manishearth** [comment]
Thanks!

File: tools/cldr-code/src/main/java/org/unicode/cldr/json/Ldml2JsonConverter.java [hunk 1]
```diff
@@ -1232,7 +1247,13 @@ public void writeReadme(String outputDir, String packageName) throws IOException
         }
         try (PrintWriter outf =
                 FileUtilities.openUTF8Writer(outputDir + "/" + packageName, "LICENSE"); ) {
-            FileCopier.copy(CldrUtility.getUTF8Data(CldrUtility.LICENSE), outf);
+            if (licenseFile.isPresent()) {
+                try (BufferedReader br = FileUtilities.openUTF8Reader("", licenseFile.get()); ) {
+                    FileCopier.copy(br, outf);
+                }
+            } else {
+                FileCopier.copy(CldrUtility.getUTF8Data("unicode-license.txt"), outf);
```
Comment (srl295) [inline tools/cldr-code/src/main/java/org/unicode/cldr/json/Ldml2JsonConverter.java:1255]: I didn't get to actually run this before.. but now that I am, it's failing.  Why was the name `unicode-license.txt` used here? That's actually _not_ in the build system, and hasn't been for some time. It should have kept the name LICENSE.  I'll reopen this and make a PR.

File: tools/cldr-code/src/main/java/org/unicode/cldr/json/Ldml2JsonConverter.java [hunk 2]
```diff
@@ -237,7 +239,15 @@ private class AvailableLocales {
                             'M',
                             "(true|false)",
                             "true",
-                            "Whether to include the -modern tier");
+                            "Whether to include the -modern tier")
+                    // Primarily useful for non-Maven build systems where CldrUtility.LICENSE may
+                    // not be available as it is put in place by pom.xml
+                    .add(
+                            "license-file",
+                            null,
```
Comment (srl295) [inline tools/cldr-code/src/main/java/org/unicode/cldr/json/Ldml2JsonConverter.java:247]: null might not be right here

**Pull request merged**